### PR TITLE
cleanup normalization, dimension_indexers, nan handling

### PR DESCRIPTION
**Pull request by 14renus**
Main changes:
- read both graphcast and pangu norm stats from .nc files (needed for internal version)
- dimension_indexers in Era5Datasets used for selecting resolution, levels as well as setting coords in convert_to_xarray()
- merge self.levels into self.dimension_indexers (no longer separate self.levels param in dataset)
- update unit tests
  - add test_normalization.py
  - merge test_era5_graphcast.py with test_era5.py to test both pangu and graphcast norm_schemes

Also makes some fixes from last PR #37 such as adding stats_cfg to:
- era5_pred.yaml and era5pred.py
- multistep_eval.py
- main_hydra.py

### CODE REVIEW

**Review by robert-DL — APPROVED** [review]
Seems fine to me, thanks for cleaning up the mess!

File: geoarches/configs/dataloader/era5.yaml [hunk 1]
```diff
@@ -3,13 +3,17 @@ dataset:
   path: data/era5_240/full/
   lead_time_hours: 24
   multistep: ${oc.select:module.train.rollout_iterations,1}
-  norm_scheme: pangu
   load_prev: True
-  variables: 
+  variables:
```
[the remark this answers is not in the document]
Comment (14renus) [reply geoarches/configs/dataloader/era5.yaml:7]: yeah good idea

File: geoarches/dataloaders/era5.py [hunk 2]
```diff
@@ -168,11 +78,10 @@ def __init__(
         domain: str = "train",
         filename_filter: Callable | None = None,
         variables: Dict[str, List[str]] | None = None,
-        levels: List[int] = None,
-        dimension_indexers: Dict[str, list] | None = None,
+        dimension_indexers: Dict[str, Tuple[str, Any]] = default_dimension_indexers,
         return_timestamp: bool = False,
         warning_on_nan: bool = True,
-        interpolate_nans: bool = True,
+        interpolate_nans: bool = False,
```
[the remark this answers is not in the document]
Comment (14renus) [reply geoarches/dataloaders/era5.py:84]: yeah good point - long run if normalization is moved, we don't need this anymore.

yeah agreed it'd be better to have a nan mask for loss and metric computation - seems cleaner and less error prone.

Also, since internal code requires model forward pass to be jittable, I had to remove usage of check_pred_has_no_nans and use a mask anyways.
Reason: we cannot have tensor values used in control flow (see [TracerBoolConversionError](https://docs.jax.dev/en/latest/errors.html#jax.errors.TracerBoolConversionError))
